scripts/collection_from_items.py

```diff
@@ -6,19 +6,92 @@
 from boto3 import client
 from linz_logger import get_log
 
-from scripts.cli.cli_helper import coalesce_multi_single
+from scripts.cli.cli_helper import coalesce_multi_single, nullable_str, valid_date
 from scripts.files.fs_s3 import bucket_name_from_path, get_object_parallel_multithreading, list_json_in_uri
 from scripts.logging.time_helper import time_in_ms
 from scripts.stac.imagery.collection import ImageryCollection
+from scripts.stac.imagery.generate_metadata import ElevationSubtypes, ImagerySubtypes, generate_description, generate_title
 from scripts.stac.imagery.provider import Provider, ProviderRole
 
 
+# pylint: disable-msg=too-many-locals
 def main() -> None:
     parser = argparse.ArgumentParser()
+    parser.add_argument("--preset", dest="preset", required=True, help="Standardised file format. Example: webp")
     parser.add_argument("--uri", dest="uri", help="s3 path to items and collection.json write location", required=True)
     parser.add_argument("--collection-id", dest="collection_id", help="Collection ID", required=True)
-    parser.add_argument("--title", dest="title", help="Collection title", required=True)
-    parser.add_argument("--description", dest="description", help="Collection description", required=True)
+    parser.add_argument(
+        "--subtype",
+        dest="subtype",
+        help="Dataset subtype description",
+        required=True,
+        choices=[
+            ImagerySubtypes.AERIAL,
+            ImagerySubtypes.HISTORICAL,
+            ImagerySubtypes.SATELLIE,
+            ImagerySubtypes.URBAN,
+            ImagerySubtypes.RURAL,
+            ElevationSubtypes.DEM,
+            ElevationSubtypes.DSM,
+        ],
+    )
+    parser.add_argument(
+        "--region",
+        dest="region",
+        help="Region of Dataset",
+        required=True,
+        choices=[
+            "antarctica",
+            "auckland",
+            "bay-of-plenty",
+            "canterbury",
+            "gisborne",
+            "global",
+            "hawkes-bay",
+            "manawatu-whanganui",
+            "marlborough",
+            "nelson",
+            "new-zealand",
+            "northland",
+            "otago",
+            "pacific-islands",
+            "southland",
+            "taranaki",
+            "tasman",
+            "waikato",
+            "wellington",
+            "west-coast",
+        ],
+    )
+    parser.add_argument("--gsd", dest="gsd", help="GSD of imagery Dataset", type=nullable_str, required=True)
+    parser.add_argument(
+        "--location", dest="location", help="Optional Location of dataset, e.g.- Hutt City", type=nullable_str, required=False
```

Comment:
So in argo, if I put in the workflow
```
- name: location
 description: '(Optional) Additional datatset description, to be used in the title in place of the Region, e.g.- Hamilton 0.05m Urban Aerial Photos (2023)'
```
The user has to enter a value otherwise they will get an error like:
` {"code":3,"message":"spec.arguments.location.value is required"}: spec.arguments.location.value is required`

In other places to get around this (i.e., to make an input parameter optional) we have used:
```
- name: location
 description: '(Optional) Additional datatset description, to be used in the title in place of the Region, e.g.- Hamilton 0.05m Urban Aerial Photos (2023)'
value: ''
```

I was under the impression that this will pass an empty string to create_collection.py instead of None.
But in this scenario `event=''` is the same as `event=None` right? so I have removed the nullable_str.

Reply:
Oh right, you're working around Argo's lack of a way to not pass empty parameter values. Sounds good!